bench/nested-deps/bench.mjs

```diff
@@ -165,35 +171,83 @@ function waitFor(millis) {
   return new Promise((resolve) => setTimeout(resolve, millis))
 }
 
-async function main() {
-  await fs.rmDir('.next', { recursive: true })
-  const file = new File('pages/index.jsx')
-  try {
+await fs.rm('.next', { recursive: true }).catch(() => {})
+const file = new File(join(CWD, 'pages/index.jsx'))
+const results = []
+
+try {
+  if (command === 'dev' || command === 'all') {
     const instance = await runNextCommandDev(['dev', '--port', '3000'])
-    const res = await fetch('http://localhost:3000/')
+
+    function waitForCompiled() {
+      return new Promise((resolve) => {
+        function waitForOnData(data) {
+          const message = data.toString()
+          const compiledRegex =
+            /compiled client and server successfully in (\d*[.]?\d+)\s*(m?s) \((\d+) modules\)/gm
+          const matched = compiledRegex.exec(message)
+          if (matched) {
+            resolve({
+              'time (ms)': (matched[2] === 's' ? 1000 : 1) * Number(matched[1]),
+              modules: Number(matched[3]),
+            })
+            instance.stdout.removeListener('data', waitForOnData)
+          }
+        }
+        instance.stdout.on('data', waitForOnData)
+      })
+    }
+
+    const [res, initial] = await Promise.all([
+      fetch('http://localhost:3000/'),
+      waitForCompiled(),
+    ])
     if (res.status !== 200) {
       throw new Error('Fetching / failed')
     }
 
-    await waitFor(3000)
+    results.push(initial)
 
     file.prepend('// First edit')
 
-    await waitFor(5000)
+    results.push(await waitForCompiled())
+
+    await waitFor(1000)
 
     file.prepend('// Second edit')
 
-    await waitFor(5000)
+    results.push(await waitForCompiled())
+
+    await waitFor(1000)
 
     file.prepend('// Third edit')
 
-    await waitFor(5000)
+    results.push(await waitForCompiled())
+
+    console.table(results)
 
     await killApp(instance)
-    await fs.rmDir('.next', { recursive: true })
-  } finally {
-    file.restore()
   }
+  if (command === 'build' || command === 'all') {
+    // ignore error
+    await fs.rm('.next', { recursive: true }).catch(() => {})
```

Comment:
if we have devdeps to `rimraf` (https://github.com/vercel/next.js/pull/35306/files#diff-6cdeaa02901aa42fe219386bfceff71b3fed6ef3c42c430757d818e7f098cfbdR14), maybe we can use it here instead?

Reply:
I think `rimraf` is convenient if we want to delete files with `glob`. But scenario here is to delete a specified folder, use `rimraf` here will need to pass `{ disableGlob: true }` additionally, it's too annoyed.

Reply:
If we were to delete dir with force, we couldn't use `force` instead of suppress error?

https://nodejs.org/api/fs.html#fsrmdirpath-options-callback

> To get a behavior similar to the rm -rf Unix command, use [fs.rm()](https://nodejs.org/api/fs.html#fsrmpath-options-callback) with options { recursive: true, force: true }.

Reply:
Sure, I'll add `force: true` here.